libs/net/dal/Services/ContentService.cs

```diff
@@ -150,6 +160,249 @@ public IPaged<Content> Find(ContentFilter filter, bool asNoTracking = true)
+            var publishedStartOn = filter.PublishedStartOn.Value.ToUniversalTime().ToString("s") + "Z";
+            var publishedEndOn = filter.PublishedEndOn.Value.ToUniversalTime().ToString("s") + "Z";
+            filterQueries.Add(s => s.DateRange(m => m
+                .Field(p => p.PublishedOn)
+                .GreaterThanOrEquals(publishedStartOn)
+                .LessThanOrEquals(publishedEndOn)));
+        }
+        else if (filter.PublishedStartOn.HasValue)
+        {
+            var publishedStartOn = filter.PublishedStartOn.Value.ToUniversalTime().ToString("s") + "Z";
+            filterQueries.Add(s => s.DateRange(m => m
+                .Field(p => p.PublishedOn)
+                .GreaterThanOrEquals(publishedStartOn)));
+        }
+        else if (filter.PublishedEndOn.HasValue)
+        {
+            var publishedEndOn = filter.PublishedEndOn.Value.ToUniversalTime().ToString("s") + "Z";
+            filterQueries.Add(s => s.DateRange(m => m
+                .Field(p => p.PublishedOn)
+                .LessThanOrEquals(publishedEndOn)));
+        }
+
+        if (filter.UserId.HasValue)
+        {
+            filterQueries.Add(s => s.Term(t => t.OwnerId, filter.UserId.Value));
+        }
+
+        if (filter.Status.HasValue)
+        {
+            filterQueries.Add(s => s.Match(m => m.Field(p => p.Status).Query(filter.Status.Value.ToString())));
+        }
+
+        var response = await _client.SearchAsync<Content>(s => {
+            var result = s
+                .Pretty()
+                .Index(_unpublishedIndex)
+                .From((filter.Page - 1) * filter.Quantity)
+                .Size(filter.Quantity)
+                .Query(q => q.Bool(b => b
+                    .Filter(filterQueries)
+                    .Should(shouldQueries)))
+                ;
+
+            if (filter.Sort.Any())
+            {
+                Expression<Func<Content, dynamic>>? objPath = null;
+                var sorts = filter.Sort[0];
+                var sort = sorts.Split(' ')[0];
+
+                if (sort == "id") objPath = p => p.Id;
+                if (sort == "productId") objPath = p => p.ProductId;
+                if (sort == "ownerId") objPath = p => p.OwnerId!;
+                if (sort == "publishedOn") objPath = p => p.PublishedOn!;
+                if (sort == "otherSource") objPath = p => p.OtherSource.Suffix("keyword");
+                if (sort == "page") objPath = p => p.Page.Suffix("keyword");
+                if (sort == "status") objPath = p => p.Status.Suffix("keyword");
+
+                if (objPath != null) result = result.Sort(s => sorts.EndsWith(" desc") ? s.Descending(objPath) : s.Ascending(objPath));
+            }
+            else
+            {
+                result = result.Sort(s => s.Descending(p => p.PublishedOn).Descending(p => p.Id));
+            }
+
+            return result;
+        });
+
+        var items = response.IsValid ? response.Documents : Array.Empty<Content>();
```

Comment:
What is an invalid response?  Should it be hidden by an empty array?  It's more likely it needs to bubble out to inform the user.

Reply:
Updated, thanks